core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -191,6 +227,100 @@ oppia.controller('LearnerDashboard', [
       return value;
     };
 
+    $scope.sortFeedbackFunction = function(entity) {
+      return entity[$scope.currentFeedbackThreadsSortType];
+    }
+
+    $scope.onClickThread = function(
+      explorationId, threadId, explorationTitle) {
+      var threadDataUrl = (
+        '/learnerdashboardthreadhandler/' + explorationId + '/' + threadId);
+      $scope.threadTitle = explorationTitle;
+      $scope.feedbackThreadActive = true;
+      $scope.explorationId = explorationId;
+      $scope.threadId = threadId;
+
+      for (var index = 0; index < $scope.threadSummaries.length; index++) {
+        if ($scope.threadSummaries[index].exploration_id === explorationId &&
+            $scope.threadSummaries[index].thread_id === threadId) {
+          $scope.threadIndex = index;
+          var threadSummary = $scope.threadSummaries[index]
+          if (typeof threadSummary.second_last_message_read !== 'undefined') {
+            $scope.threadSummaries[index].second_last_message_read = true;
+          }
+          if (threadSummary.last_message_read === false) {
+            $scope.numberOfUnreadThreads -= 1;
+          }
+          threadSummary.last_message_read = true;
+        }
+      }
+
+      $http.get(threadDataUrl).then(function(response) {
+        var data = response.data;
+        $scope.messageSummaries = data.message_summary_list;
+      });
+    };
+
+    $scope.backFromThread = function() {
+      $scope.feedbackThreadActive = false;
+    };
+
+    $scope.addNewMessage = function(explorationId, threadId, newMessage) {
+      var url = '/threadhandler/' + explorationId + '/' + threadId;
```

Comment:
I had earlier used this but then I get an error message like this:
![image](https://user-images.githubusercontent.com/13691317/28447804-e3da39ca-6df0-11e7-8c5e-3a65834b5365.png).
The thread id contains characters that are not allowed. What should we do about this?